tachyon/crypto/hashes/sponge/poseidon/poseidon_sponge_base.h

```diff
@@ -27,13 +27,73 @@ struct PoseidonSpongeBase : public FieldBasedCryptographicSponge<Derived> {
     auto& config = derived.config;
     if (is_full_round) {
       // Full rounds apply the S-Box (xᵅ) to every element of |state|.
-      for (F& elem : state.elements) {
-        elem = elem.Pow(config.alpha);
+      if (config.alpha == 3) {
+        for (F& elem : state.elements) {
+          F t = elem;
+          // e²
+          t.SquareInPlace();
+          // e³
+          elem *= t;
```

Comment:
```suggestion
        for (F& e : state.elements) {
          F t = e;
          // e²
          t.SquareInPlace();
          // e³
          e *= t;
```
elem -> e
or
how about removing the comment?

Reply:
I'll remove the comment!